### PR TITLE
Cloudish scheduler refactoring

### PR DESCRIPTION
**Pull request by jvmafra**
The refactoring includes:

- Decrease number of data structures, using only running and waiting lists to store requests
- Remove usage of Instance object in scheduler
- Use of request's metadata to store fields like priority, submit time and host name
- Save to a file the scheduler status (Lists of running and waiting vms) and get it when restarting the service
- Refactoring the architecture to facility the implementation of new preemption policies

### CODE REVIEW

**Comment by marcuswac** [comment]
There are many changes that are not needed (blank spaces, line breaks, etc).

It would be good to have a style checker (I think OpenStack uses flake8) to run on every PR. I think we need a Continuous Integration tool for that (OpenStack uses Jenkins).

For this PR, take another look in this blank spaces and line breaks that are not needed.

File: nova/scheduler/__init__.py [hunk 1]
```diff
@@ -19,4 +19,4 @@
 .. automodule:: nova.scheduler
    :platform: Unix
    :synopsis: Module that picks a compute node to run a VM instance.
-"""
+"""
```
Comment (marcuswac) [inline nova/scheduler/__init__.py]: This change is not needed.

File: nova/scheduler/host_manager.py [hunk 2]
```diff
@@ -176,14 +176,14 @@ def _locked_update(self, compute, service, aggregates, inst_dict):
             if inst_dict is not None:
                 LOG.debug("Update host state with instances: %s", inst_dict)
                 self.instances = inst_dict
-                print "Updating..."
-                print "Len instances = %d" % len(self.instances)
+                # print "Updating..."
```
Comment (marcuswac) [inline nova/scheduler/host_manager.py]: Do we need to comment these prints? If not, remove them.
Let's change only things that are really needed.

File: nova/scheduler/instance_manager.py [hunk 3]
```diff
@@ -1,99 +0,0 @@
-from enum import Enum
```
Comment (marcuswac) [inline nova/scheduler/instance_manager.py]: This file was removed? Why?

File: nova/scheduler/manager.py [hunk 4]
```diff
@@ -139,7 +139,6 @@ def delete_instance_info(self, context, host_name, instance_uuid):
 
         self.driver.host_manager.delete_instance_info(context, host_name,
                                                       instance_uuid)
-
```
Comment (marcuswac) [inline nova/scheduler/manager.py]: This change is not needed.

File: nova/scheduler/preemption_utils.py [hunk 5]
```diff
@@ -1,116 +0,0 @@
-from datetime import datetime
```
Comment (marcuswac) [inline nova/scheduler/preemption_utils.py]: What happened with this file?

File: nova/scheduler/instance_manager.py (hunk 3 above)
Comment (jvmafra) [inline nova/scheduler/instance_manager.py]: We're only using RequestSpec object to manage the Vms. Attributes like submit time and priority are in metadata of RequestSpec object now

File: nova/scheduler/preemption_utils.py (hunk 5 above)
Comment (jvmafra) [inline nova/scheduler/preemption_utils.py]: The methods to calculate free space on hosts are included in Preemption Policy class

**Pull request merged**